mParticle-Apple-SDK/Listener/MPListenerController.swift

```diff
@@ -0,0 +1,353 @@
+     * {@see MPConsentState} required it to be stopped, the Kit crashed, or a
+     * configuration was received that excluded the kit
+     * @param kitId the id of the kit, corresponds with a {@see MPKitInstance}
+     * @param reason a message containing the reason a kit was stopped
+     */
+    @objc optional func onKitExcluded(_ kitId: Int32, reason: Any!)
+
+    /**
+     * Indicates that state of a Session may have changed
+     * @param session the current {@see MParticleSession} instance
+     */
+    @objc optional func onSessionUpdated(_ session: Any!)
+
+    /**
+     * Indicates that an alias request has completed
+     * @param aliasResponse the alias response object
+     */
+    @objc optional func onAliasRequestFinished(_ aliasResponse: MPAliasResponse?)
+}
+
+
+@objc public class MPListenerController : NSObject {
+    private static let shared = MPListenerController()
+    private var sdkListeners: [MPListenerProtocol] = []
+
+    /**
+     * Returns the shared instance object.
+     * @returns the Singleton instance of the MPListener class.
+     */
+    @objc public class func sharedInstance() -> MPListenerController {
+        return shared
+    }
+
+    /**
+     * Adds a listener to the SDK to receive any MPListenerProtocol calls from the API to that object
+     * @param sdkListener An instance of a class that implements the MPListenerProtocol
+     */
+    @objc public func addSdkListener(_ sdkListener: any MPListenerProtocol) {
+        self.sdkListeners.append(sdkListener)
+    }
+
+    /**
+     * Removes a listener from the SDK to no longer receive any MPListenerProtocol calls from the API to that object
+     * If you don't remove the Listener we will retain a zombie reference of your object and it will never be released
+     * @param sdkListener An instance of a class that implements the MPListenerProtocol
+     */
+    @objc public func removeSdkListener(_ sdkListener: any MPListenerProtocol) {
+        self.sdkListeners = self.sdkListeners.filter { sdkListener !== $0 }
+    }
+
+    /**
+     * Indicates that an API method was called. This includes invocations both from external sources (your code)
+     * and those which originated from within the SDK
+     * @param apiName the name of the API method
+     * @param parameter1 to parameter4 are the arguments sent to this api, such as the MPEvent in logEvent
+     */
+    @objc public func onAPICalled(_ apiName: Selector, parameter1: NSObject?, parameter2: NSObject?, parameter3: NSObject?, parameter4: NSObject?) {
+        
+    }
+
+    @objc public func onAPICalled(_ apiName: Selector, parameter1: NSObject?, parameter2: NSObject?, parameter3: NSObject?) {
+        for delegate in self.sdkListeners {
+            let stackTrace = [Thread.callStackSymbols]
+            var parameters: [NSObject] = []
+            if parameter1 != nil {
+                parameters.append(parameter1!)
```

Comment:
```suggestion
            if let parameter1 = parameter1 {
                parameters.append(parameter1)
```

All of these nil checks should be replaced with `if let` and all uses of the `!` operator removed.